src/edge/simulator.ts

```diff
@@ -226,10 +196,156 @@ export class Simulator {
+                    version = releaseInfoJson.tag_name;
+                }
+            } else {
+                const pipResponse: string = await request.get(Simulator.iotedgehubdevVersionUrl);
+                version = JSON.parse(pipResponse).info.version;
+            }
+            return version;
+        } catch (error) {
+            return undefined;
+        }
+    }
+
+    private getUserConfiguredVersion(): string {
+        const version: string = Configuration.getConfigurationProperty("simulator.version");
+        if (version && version.match(/^\d+\.\d+\.\d+$/)) {
+            return "v" + version;
+        }
+    }
+
+    private async getCurrentSimulatorVersion(): Promise<string | undefined> {
+        const output: string = await Executor.executeCMD(undefined, this.getSimulatorExecutorPath(true), { shell: true }, "--version");
+        const version: string | null = Simulator.extractVersion(output);
+        return version;
+    }
+
+    private async simulatorInstalled(): Promise<boolean> {
+        return await Simulator.checkCmdExist(this.getSimulatorExecutorPath(true));
+    }
+
+    private getSimulatorExecutorPath(forceUseCmd: boolean = false): string {
+        let executorPath: string = "iotedgehubdev";
+        if (os.platform() === "win32") {
+            let installedVersion: string = this.context.globalState.get(Simulator.SimulatorVersionKey);
+
+            const userConfiguredVersion: string = this.getUserConfiguredVersion();
+            if (userConfiguredVersion) {
+                installedVersion = userConfiguredVersion;
+            }
+
+            const simulatorPath: string = path.join(Simulator.WindowsStandaloneSimulatorFolder, installedVersion , executorPath);
+
+            if (!forceUseCmd) {
+                executorPath = `"${Utility.adjustFilePath(simulatorPath)}"`;
+                if (Utility.isUsingPowershell()) {
+                    executorPath = `& ${executorPath}`;
+                }
+            } else {
+                executorPath = `"${simulatorPath}"`;
+            }
+        }
+        return executorPath;
+    }
+
+    private async downloadStandaloneSimulatorWithProgress() {
+        await vscode.window.withProgress({
+            location: vscode.ProgressLocation.Notification,
+            title: "Downloading and installing Azure IoT EdgeHub Dev Tool (iotedgehubdev)...",
+        }, async () => {
+            await this.downloadStandaloneSimulator();
+        });
+    }
+
+    private async downloadStandaloneSimulator() {
+        const userConfiguredVersion: string = this.getUserConfiguredVersion();
+        let version: string;
+        let binariesZipUrl: string;
+        if (userConfiguredVersion) {
+            version = userConfiguredVersion;
+            binariesZipUrl = `https://github.com/Azure/iotedgehubdev/releases/download/${version}/iotedgehubdev-${version}-win32-ia32.zip`;
```

Comment:
create a short link for the github release url

Reply:
Use a static  variable for root url :     
```
private static iotedgehubdevDownloadRootUrl = "https://github.com/Azure/iotedgehubdev/releases/download";
```